### ISSUE TITLE
Some issues with notifications and sent messages

### ISSUE THREAD

**Issue by Sacolevas** [emesene/emesene#1362]
Hello,

Because of the segfault problem in 2.12.5, I had recently downgraded to 2.12.3, but still I have found 
some problems that also have in the latest version, so I guess they correspond to some issues that 
have not been fixed yet. 

First, some msn contacts have told me that they don't see me online, even if I choose the "online" 
option all the time (and I don't use the idle status extension). And sometimes they don't receive my 
messages or I don't receive their answers. So, I am trying to check errors in a normal season to help you find 
an explanation:

Error 1: 

I tried to send a message to a friend and then some lines after, the following error appeared. Does this mean my 
message was not sent?

---

[09:12:58 DEBUG papyon.transport] >>> MSG 13 D 250
    Content-Type: 'application/x-msnmsgrp2p'\r\n
    MIME-Version: '1.0'\r\n
    P2P-Src: 'sender@live.com;{d82bcfa8-b6d9-4f7c-891c-32de8bc03743}'\r\n
    P2P-Dest: 'receiver@hotmail.com;{babd8946-f598-4731-85ba-798554bd09e1}'\r\n
    [P2P message (16 bytes)]
[09:12:58 ERROR papyon.msnp2p.session_manager] No handler could handle euf-guid {A4268EEC-FEC5-49e5-95C3-F126696BDBF6}

---

Error 2: 

I installed all the normal stuff in Ubuntu and dependences, notification-daemon, etc. But 
the following error appears rather frequently. In my preferences, I have the default configuration in everything, with the exception that I ckecked the see the "tray icon" and I checked the option to "notify when a contact is typing"

What does this error mean?

---

[09:23:49 ERROR e3.common.Signal] Signal handler (_on_user_typing._on_user_typing) error: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.Notifications was not provided by any .service files
Traceback (most recent call last):
  File "/usr/share/emesene/emesene/e3/common/Signal.py", line 65, in emit
    if callback(_args, *_kwargs) == False:
  File "/usr/share/emesene/emesene/e3/common/Signal.py", line 112, in __call__
    self.f(self.c(), *arg)
  File "/usr/share/emesene/emesene/e3/common/notification.py", line 110, in _on_user_typing
    self._notify(contact, contact.nick, _("is typing a message"), account)
  File "/usr/share/emesene/emesene/e3/common/notification.py", line 200, in _notify
    self.notifier(title if title else '', text, uri, 'message-im', None, tooltip)
  File "/usr/share/emesene/emesene/gui/common/PyNotification.py", line 47, in PyNotification
    notification.show()
GError: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.Notifications was not provided by any .service files

---

Similar error: 

[09:47:55 ERROR e3.common.Signal] Signal handler (_on_message._on_message) error: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.Notifications was not provided by any .service files
Traceback (most recent call last):
  File "/usr/share/emesene/emesene/e3/common/Signal.py", line 65, in emit
    if callback(_args, *_kwargs) == False:
  File "/usr/share/emesene/emesene/e3/common/Signal.py", line 112, in __call__
    self.f(self.c(), *arg)
  File "/usr/share/emesene/emesene/e3/common/notification.py", line 138, in _on_message
    self._notify(contact, msgobj.display_name, text, msgobj.account)
  File "/usr/share/emesene/emesene/e3/common/notification.py", line 200, in _notify
    self.notifier(title if title else '', text, uri, 'message-im', None, tooltip)
  File "/usr/share/emesene/emesene/gui/common/PyNotification.py", line 47, in PyNotification
    notification.show()
GError: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.Notifications was not provided by any .service files

---

Error 3: 

I was tryng to check which extensions I have installed, if this has relation with the previous error and the following 
error appeared:

---

 ERROR e3.common.Collections] [Errno socket error] [Errno 8] _ssl.c:504: EOF occurred in violation of protocol
Traceback (most recent call last):
  File "/usr/share/emesene/emesene/e3/common/Collections.py", line 181, in fetch_metadata
    rq = self.github.get_raw(self.theme, current_ext.files[path])
  File "/usr/share/emesene/emesene/e3/common/Github.py", line 42, in get_raw
    response = urlopen(API_GITHUB_GETRAW % (self._org, repo, sha))
  File "/usr/lib/python2.7/urllib.py", line 86, in urlopen
    return opener.open(url)
  File "/usr/lib/python2.7/urllib.py", line 207, in open
    return getattr(self, name)(url)
  File "/usr/lib/python2.7/urllib.py", line 436, in open_https
    h.endheaders(data)
  File "/usr/lib/python2.7/httplib.py", line 954, in endheaders
    self._send_output(message_body)
  File "/usr/lib/python2.7/httplib.py", line 814, in _send_output
    self.send(msg)
  File "/usr/lib/python2.7/httplib.py", line 776, in send
    self.connect()
  File "/usr/lib/python2.7/httplib.py", line 1161, in connect
    self.sock = ssl.wrap_socket(sock, self.key_file, self.cert_file)
  File "/usr/lib/python2.7/ssl.py", line 381, in wrap_socket
    ciphers=ciphers)
  File "/usr/lib/python2.7/ssl.py", line 143, in __init__
    self.do_handshake()
  File "/usr/lib/python2.7/ssl.py", line 305, in do_handshake
    self._sslobj.do_handshake()
IOError: [Errno socket error] [Errno 8] _ssl.c:504: EOF occurred in violation of protocol

---

Thanks for your attention and possible answer about the meaning of these errors. Hopefully, this report will be useful for the developers (or perhaps I am reporting issues already fixed). Thanks again :)

**Comment by Sbte** [comment]
Error 1 is harmless.

Error 2 tells you that your pynotify doesn't work/is not installed.

Error 3 is interesting. No idea what happens there

**Comment by c10ud** [comment]
3 is probably some server fault

**Comment by Sbte** [comment]
@c10ud probably, but errors are not so nice

**Comment by c10ud** [comment]
indeed, btw google suggests also that multiple newlines (\n) could trigger this, but i think this is not the case otherwise you would get that too

**Comment by Sacolevas** [comment]
Thanks for the info in error 1. In error 2, does this affects that people won't receive messages (or I don't receive all answers by my contacts?)? Friends have told me that the write to me and I don't receive all messages and I have no idea why (and I always enter as available and those people I mention have not been blocked).

I have installed python-notify, notification-daemon, etc (related programs) in a fresh install of Ubuntu 12.04, but the same error appears. I tried to reinstall and still no difference. 

I don't know how to solve this, or this is not something I can control and as you can see in issues 1356 and 1364, more people have the same issue. In issue 1364 someone realizes this is not precisely a problem of emesene, but ubuntu users have experienced it, so we don't know how to fix it and more importantly, know if this error is harmless and does not compromise the ability of receiving and sending all messages. 

Thanks for your attention

**Comment by moneycat** [comment]
@Sacolevas try this method c10ud suggest in #1364: `$ python -c 'import pynotify; pynotify.init("thisisatesttest"); n=pynotify.Notification("hello", "test", ""); n.set_hint_string("append", "allowed"); n.show()'`

If the problem is still there, then that may be an upstream problem of os platform.

**Comment by Sacolevas** [comment]
Hello,

The answer of doing that was the following:

---

Traceback (most recent call last):
File "string", line 1, in module

## glib.GError: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.Notifications was not provided by any .service files

So, I don't know if I can "live with that" (i.e being able to receive all messages from my contacts and my contact being able to receive my messages), or not. I have an almost fresh install installed the normal ubuntu packages (nothing strange), so I still don't know what to do. Thanks

**Comment by c10ud** [comment]
your OS is somehow messed up but i don't know where to point, you'd
better take a look at modifications you made to your system

2012/6/25 Sacolevas
reply@reply.github.com:

**Comment by moneycat** [comment]
@Sacolevas That is a bug of upstream (maybe gnome). I'm using Ubuntu 12.04 64-bit with gnome-shell but I did not get this error. My another gentoo box either got this.

Maybe you can ask for help in launchpad for more detail help from ubuntu: https://answers.launchpad.net/ubuntu . I hope this may help to solve your problem.

**Comment by moneycat** [comment]
BTW, this is some information os my OS. I hope this can be a reference to help you to detect the real issue:
`moneycat@Nebula:~$ dpkg --list | grep notif
ii  gir1.2-notify-0.7                      0.7.5-1                                 sends desktop notifications to a notification daemon (Introspection files)
ii  libevent-1.4-2                         1.4.14b-stable-0ubuntu1                 asynchronous event notification library
ii  libevent-2.0-5                         2.0.16-stable-1                         Asynchronous event notification library
ii  libnotify-bin                          0.7.5-1                                 sends desktop notifications to a notification daemon (Utilities)
ii  libnotify0.4-cil                       0.4.0~r3032-4                           CLI library for desktop notifications
ii  libnotify4                             0.7.5-1                                 sends desktop notifications to a notification daemon
ii  libstartup-notification0               0.12-1ubuntu1                           library for program launch feedback (shared library)
ii  libzephyr4                             3.0.1-1                                 Project Athena's notification service - non-Kerberos libraries
ii  notification-daemon                    0.7.3-1                                 daemon for displaying passive pop-up notifications
ii  notify-osd-icons                       0.7                                     Notify-OSD icons
ii  python-notify                          0.1.1-3                                 Python bindings for libnotify
ii  python-pyinotify                       0.9.2-1                                 simple Linux inotify Python bindings
ii  update-notifier                        0.119ubuntu8.4                          Daemon which notifies about package updates
ii  update-notifier-common                 0.119ubuntu8.4                          Files shared between update-notifier and other packages`

**Comment by Sacolevas** [comment]
@moneycat Thanks ... I just reinstalled all these programs and now the test @c10ud mentioned before works well. So hopefully, this would help me with my emesene experience. :)

BTW, I started this issue commenting that sometimes my friends tell me that they sent me messages I didn't receive and that not all my sent messages were received.

ANOTHER RELATED ISSUE I experience is that a lot of times people who navigate in smartphones tell me that I don't appear as connected and available at times where actually I was connected and I login as available (I always do as available). What is happening there?

Thanks!!

**Comment by c10ud** [comment]
"I had recently downgraded to 2.12.3, but still I have found
some problems that also have in the latest version, so I guess they correspond to some issues that
have not been fixed yet. "

no way, only proved and reproducible bugs for latest git head here, sorry

"BTW, I started this issue commenting that sometimes my friends tell me that they sent me messages I didn't receive and that not all my sent messages were received."
should be fixed in .12.5 or with latest change in git head (thanks to pepeleproso)

"ANOTHER RELATED ISSUE I experience is that a lot of times people who navigate in smartphones tell me that I don't appear as connected and available at times where actually I was connected and I login as available (I always do as available). What is happening there?" 
what about emesene isn't the issue here? if you show me a screenshot where you're not online (with a semiproper client, like wlm or pidgin) but you really are, i'll believe you
